transfer.py

```diff
@@ -0,0 +1,115 @@
+from kardashigans.analyze_model import AnalyzeModel
+from kardashigans.experiment import Experiment
+from random import sample
+
+
+class SplitDataset:
+    """
+    Dataset with samples corresponding to certain tagging labels only
+    """
+
+    def __init__(self, dataset, labels):
+        (x_train, y_train), (x_test, y_test) = dataset.load_data()
+        self._x_train, self._y_train = self.split_set(x_train, y_train, labels)
+        self._x_test, self._y_test = self.split_set(x_test, y_test, labels)
+
+    @staticmethod
+    def split_set(x, y, labels):
+        indices = [i for i in range(y.shape[0]) if y[i] in labels]
+        return x[indices, ...], y[indices, ...]
+
+    def load_data(self):
+        return (self._x_train, self._y_train), (self._x_test, self._y_test)
+
+
+class TransferExperiment(Experiment):
+    """
+    replicated the method demonstrated in "How transferable are features in deep neural networks?"
+    supports only trainers with "layers_to_freeze" and "weight_map" arguments
+    """
+
+    def __init__(self, dataset, trainer_class, trainer_kwargs, n_classes,
+                 split_labels=None, *args, **kwargs):
+        if split_labels is None:
+            split_labels = sample(list(range(n_classes)), n_classes // 2)
+        data_a = SplitDataset(dataset, split_labels)
+        data_b = SplitDataset(dataset, [i for i in range(n_classes) if i not in split_labels])
+        trainer_a = trainer_class(dataset=data_a, **trainer_kwargs)
+        trainer_b = trainer_class(dataset=data_b, **trainer_kwargs)
+        super(TransferExperiment, self).__init__(model_names=["a", "b"],
+                                                 trainers={
+                                                     "a": trainer_a,
```

Comment:
Maybe more descriptive model names...? Calling `self.open_model` may initiate training, and then the model is saved using the `model_name` argument (the output filename will be `a.h5`)

In any case you use the strings `"a","b"` at least twice so I would put the model names in named variables

Reply:
it's descriptive once you read the paper :) I'll added a variable. @dorimedini